akka-persistence/src/main/scala/akka/persistence/fsm/PersistentFSM.scala

```diff
@@ -54,6 +54,19 @@ trait PersistentFSM[S <: FSMState, D, E] extends PersistentActor with Persistent
   private var currentStateTimeout: Option[FiniteDuration] = None
 
   /**
+   * "akka.persistence.fsm.snapshot-after"
+   */
+  private val snapshotAfter: Option[Int] = {
+    val config = context.system.settings.config
+    val key = "akka.persistence.fsm.snapshot-after"
+
+    config.getString(key).toLowerCase match {
+      case "off" ⇒ None
+      case _     ⇒ Some(config.getInt(key))
+    }
+  }
```

Comment:
hm, what shall we do about the bin compat issue?
reading from the config each time is not an option.
placing it in an Akka extension would be possible, some overhead of doing the extension lookup each time but that is probably the best I can think of